include/flang/lower/OpBuilder.h

```diff
@@ -115,24 +116,74 @@ class CharacterOpsBuilder : public OpBuilderWrapper {
   CharacterOpsBuilder(OpBuilderWrapper &b) : OpBuilderWrapper{b} {}
   /// Interchange format to avoid inserting unbox/embox everywhere while
   /// evaluating character expressions.
-  struct CharValue {
-    fir::ReferenceType getReferenceType();
-    fir::CharacterType getCharacterType();
 
-    mlir::Value reference;
+  /// A readable character entity
+  struct CharValue {
+    /// Get fir.char<kind> type with the same kind as value.
+    fir::CharacterType getCharacterType() const;
+    /// Get fir.ref<fir.char<kind>> type.
+    fir::ReferenceType getReferenceType() const;
+    /// Get fir.sequence<? x fir.char<kind>> type.
```

Comment:
Since there is an array standard type, I called it "SequenceType" in the code. Syntactically, it is spelled as `!fir.array<>` in the IR.

Reply:
I got confused copy pasting, thanks.